java-frontend/src/main/java/org/sonar/plugins/java/api/CheckRegistrar.java

```diff
@@ -171,6 +172,16 @@ public void registerMainSharedCheck(JavaCheck check, Collection<RuleKey> ruleKey
       // to be overridden
     }
 
+    /**
+     * Registers a custom file scanner not related to any rule or repository.
+     * CheckRegistrars call this function to register a custom file scanner for execution during the analysis
+     * on all source files that match the given rule scope (MAIN, TEST or ALL).
+     * Custom file scanners reporting an issue will have no effect, since no rule is associated.
```

Comment:
`will have no effect`: meaning that it is safe to report and not trigger any issue at runtime?